docs/scripts/add_version.py

```diff
@@ -0,0 +1,94 @@
+"""Utilities for adding versions when building the documentation."""
+
+import sys
+from pathlib import Path
+
+
+def add_version_to_selector_page(version: str) -> None:
+    """Add the newly built version to the version selection overview.
+
+    Args:
+        version: The version that should be added.
+    """
+    indent = "        "
+    new_line = (
+        f"{indent}<li><a href="
+        f'"https://avhopp.github.io/baybe_dev/{version}/">{version}'
+        "</a></li>\n"
+    )
+    file = Path("versions.html")
+    modified_lines = []
+    with file.open(mode="r") as f:
+        lines = f.readlines()
+        for line in lines:
+            modified_lines.append(line)
+            # Add new line at correct position which is in the first line after stable
```

Comment:
Just a remark: I guess your approach would break in case we release a version out of the regular order? That is, when we are already on version 0.14.0 but want to nevertheless bring a hotfix to version 0.13.4 using a new tag 0.13.5

Reply:
Depends on what you mean by "breaking". The order on the versions page would break, and the newest version would automatically be labeled as "stable". But I have to say that I struggle to see the reason for making hotfixes to old versions unless there are breaking changes preventing people from just switching to a higher version.

Reply:
My suggestion (and also @Scienfitz please share your opinion here): We do not include this case now. Currently, it is purely hypothetical whether we will ever have the case of such a hotfix, and it would require some re-thinking of how to automatically build everything.

My approach would be: As soon as we have the case of "We need to create a fix for an old version and push the code and documentation for this, but this should not be the new stable", we fix this workflow.

Also, one of the advantages of this new approach of doc building is that we can easily manually include such new documentations/fix errors since everything is now just organized via folders on a branch, so in the absolute worst case, we can always just manually push stuff to the branch and fix the documentation.

Reply:
Hey @AVHopp, I'm fine with your plan 👍🏼 I just want to point out that the situation I mentioned (i.e. you need to publish a hotfix for an older version) is perhaps way less exotic that you may think – it just hasn't happened yet because there is still a manageable number of projects where baybe is used and nothing is critical for production. As soon as things get a little tougher (which could simply be more projects or projects closer to industrialization) this will look very differently.

So what I'm trying to say: I see this situation quite certainly coming, but a few months down the road. So be prepared that we quite certainly need to touch this again. Hence, I'll leave it up to you to decide whether you want to take care of it now (where you a still concerned with it) or want to be pinged in 6 months where I call you and say "Hey @AVHopp, we **urgently** (which is the nature of hotfixes) need to release a hotfix for version x.y.z, so can you please add this missing piece in the next three hours?"

Reply:
would it be possible to extend the logic here a bit and find the right line position? essentially a sorting

Reply:
Done.